pkg/webhook/admission/pod/agent_injector.go

```diff
@@ -196,19 +197,52 @@ func (ag *AgentInjector) InjectAgent(pod *v1.Pod) error {
 	}
 
 	queueProxyEnvs := []v1.EnvVar{}
-	queueProxyAvailable := false
+	containerPort := constants.InferenceServiceDefaultHttpPort
+	readinessProbeAvailable := false
+
 	for _, container := range pod.Spec.Containers {
 		if container.Name == "queue-proxy" {
 			queueProxyEnvs = container.Env
-			queueProxyAvailable = true
+		}
+
+		if container.Name == "kserve-container" {
+			containerPort = fmt.Sprint((container.Ports[0].ContainerPort))
+			args = append(args, "--component-port", containerPort)
 		}
 	}
 
-	if !queueProxyAvailable {
-		readinessProbeJson, err := json.Marshal(pod.Spec.Containers[0].ReadinessProbe)
-		if err != nil {
-			return err
+	containerReadinessProbe := &v1.Probe{
+		Handler: v1.Handler{
+			TCPSocket: &v1.TCPSocketAction{
+				Port: intstr.FromString(containerPort),
+			},
+		},
+		TimeoutSeconds:   1,
+		PeriodSeconds:    10,
+		SuccessThreshold: 1,
+		FailureThreshold: 3,
+	}
+
+	readinessProbeJson, err := json.Marshal(containerReadinessProbe)
+	if err != nil {
+		return err
+	}
+
+	for i, envVar := range queueProxyEnvs {
+
+		if envVar.Name == "SERVING_READINESS_PROBE" {
+			readinessProbeAvailable = true
+			envVar.Value = string(readinessProbeJson)
```

Comment:
So this will overwrite the user defined readiness probe?

Reply:
Not exactly sure, looks like it from the code. Let me check, should be overriding only when default probe is attached.

Reply:
Here actually mutates the queue proxy's `SERVING_READINESS_PROBE` and `USER_PORT` as `queueProxyEnvs` holds a reference to the queue proxy's env vars. As we are now inject agent between `queue-proxy` and `kserve-container`, `queue-proxy` should send probes to agent while agent sends probes to `kserve-container`.

Reply:
We can take a copy of `queueProxyEnvs` and that should be what's set on the agent container before we mutate the `queue-proxy` envs